src/main/kotlin/com/sourcegraph/cody/chat/ui/LlmDropdown.kt

```diff
@@ -52,12 +53,17 @@ class LlmDropdown(
 
     models.filterNot { it.deprecated }.sortedBy { it.codyProOnly }.forEach(::addItem)
 
-    val selectedFromState = chatModelProviderFromState
-    val selectedFromHistory = HistoryService.getInstance(project).getDefaultLlm()
-    selectedItem =
-        models.find { it.model == selectedFromState?.model && !it.deprecated }
-            ?: models.find { it.model == selectedFromHistory?.model && !it.deprecated }
-            ?: models.find { it.default }
+    CodyAuthenticationManager.getInstance(project).getActiveAccountTier().thenApply { accountTier ->
+      if (accountTier == null) return@thenApply
+      isCurrentUserFree = accountTier == AccountTier.DOTCOM_FREE
```

Comment:
`getActiveAccountTier` is cached, we could use it directly in renderer instead of caching it second time there.